bitcoin/src/blockdata/transaction.rs

```
@@ -391,6 +397,26 @@ impl TransactionExt for Transaction {
         self.input.len() == 1 && self.input[0].previous_output == OutPoint::COINBASE_PREVOUT
     }
 
+    fn coinbase_height(&self) -> Result<u64, Bip34Error> {
+        if !self.is_coinbase() {
+            return Err(Bip34Error::NotPresent);
+        }
+        
+        let input = self.input.first().ok_or(Bip34Error::NotPresent)?;
+        let push = input
+            .script_sig
+            .instructions_minimal()
+            .next()
+            .ok_or(Bip34Error::NotPresent)?
+            .map_err(to_bip34_error)?;
+        
+        match (push.script_num(), push.push_bytes().map(|b| b.read_scriptint())) {
```

Didn't we add number reading to `Instructions` already?

---

We could use `read_int` but it internally swallows errors like `NonMinimalPush` via `.ok()`, which causes some tests to fail. 

I think keeping the explicit handling with`read_scriptint()` is a bit more verbose but gives us the right behavior

---

Not failing tests shouldn't be the goal but in this case it seems like `read_int` should return `Result` instead.

It's not about verbosity but avoiding duplication of code that ought to do the same thing.